code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -76,6 +76,9 @@
 	if(!affecting) //missing limb? we select the first bodypart (you can never have zero, because of chest)
 		affecting = bodyparts[1]
 	send_item_attack_message(I, user, affecting.name)
+	if(reagents)
+		for(var/datum/reagent/R in reagents.reagent_list)
+			R.on_mob_attacked(src)

```

Comment:
Please implement the retaliation in a more generic manner, preferably as a component that is added and removed from the mob by the reagent. I very much dislike this pattern of implementing functionality with no-op loops like this, this will trigger a retaliation maybe one time in twenty thousand iterations if that, most likely never in a round.

Reply:
The idea here was to make a generic handler for any reagent to have an effect that triggers when the mob is attacked, for future reagents that might want to implement on-attack procs. If you don't like that, I can certainly change it to something more specific to stabsinthe itself

Reply:
components would be able to do that

Reply:
Alright, I'll take a look! It might take a little bit, I don't exactly have a full idea of what I'm doing